pkg/kubelet/kubelet.go

```diff
@@ -2513,6 +2527,17 @@ func (kl *Kubelet) syncLoopIteration(ctx context.Context, configCh <-chan kubety
 			}
 			klog.V(4).InfoS("SyncLoop (housekeeping) end", "duration", duration.Round(time.Millisecond))
 		}
+	case <-kl.nodeRegistrationCh:
+		// It is possible that static pods are running but their corresponding mirror
+		// pods are not created because node was not registered. We want to sync static
+		// pods again quickly on node registration so that their resource requests are
+		// visible to the scheduler and avoid delaying node readiness.
+		if kl.kubeClient != nil {
+			staticPods, mirrorPods := kl.podManager.GetStaticPodsAndMirrorPods()
+			if len(staticPods) != len(mirrorPods) {
+				handler.HandlePodSyncs(staticPods)
```

Comment:
It's probably not good to use the main thread for creating pods at the apiserver. 

Another possible approach is to move the mirror pod creation to the beginning of the syncpod to accelerate the operations.

This would also help minimize the window for race condition for any new static pods that are added after node registration.